polkadot/xcm/xcm-executor/src/lib.rs

```diff
@@ -1002,13 +1002,18 @@ impl<Config: config::Config> XcmExecutor<Config> {
 			InitiateReserveWithdraw { assets, reserve, xcm } => {
 				let old_holding = self.holding.clone();
 				let result = Config::TransactionalProcessor::process(|| {
+					let assets = self.holding.saturating_take(assets);
+					// Must ensure that we recognise the assets as being managed by the destination.
+					#[cfg(not(feature = "runtime-benchmarks"))]
```

Comment:
just out of curiosity, which benchmarks failed because of this? Because `fn worst_case_holding` generates different stuff? If so, we can to some follow-up to remove this maybe

Reply:
yes, benchmarking code doesn't generate "trusted" assets for teleporting and reserve transferring, but that's fine for benchmarking, all assets have same/similar weight so it doesn't matter

Reply:
this have an unfortunate side effect is that we sometime could run non benchmarking tests with benchmarks feature and cusing false negative.

this is an example https://github.com/open-web3-stack/open-runtime-module-library/pull/1011/commits/7506400d700302ae4f1ee273d4e07c5f7ea6060c

the test workflow run all the tests, including benchmark one, and due to feature pollution, the runtime-benchmarks feature got added and disabled the check
the code coverage one only run tests, so without the flag, and failed the test